### PR TITLE
Team 2 projectiles - game balance and range for projectiles. 

### PR DESCRIPTION
**Pull request by brose7561**
# Description

This branch includes range for projectiles. updating weapon implementation slightly to push the config weapon stats through to where the projectile is shot in the firing controller. the values of the weapon and projectile config where updated to sort out game balance. this balance has been discussed in the studio last week and a consensus reached. however, if this implementation remains less ideal, as a group we are able to rebalance with particularly the animal team and weapon in mind. 

Fixes / Closes # (issue)
#415 
#412 
#414 

## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] tested range in weapon attack component tests and projectile actions. one ensures that ranged movement works the other assess that ranged attack works.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/entities/configs/WeaponConfig.java [hunk 1]
```diff
@@ -15,8 +15,8 @@ public class WeaponConfig {
 
     static {
         // Load weapon data (can be from JSON or hardcoded for now)
-        weaponConfigs.put("Shotgun", new WeaponData("Shotgun", 10, 10,
-                5, 20, 20, 3, "images" +
+        weaponConfigs.put("Shotgun", new WeaponData("Shotgun", 15, 3,
```
Comment (Aryan-S-Gupta) [inline source/core/src/main/com/csse3200/game/entities/configs/WeaponConfig.java:18]: The Functioning is correct, better to do it as a JSON, we might be able to change it based on difficulty level

File: source/core/src/main/com/csse3200/game/entities/factories/ProjectileFactory.java [hunk 2]
```diff
@@ -207,15 +209,24 @@ public Entity createProjectile(ProjectileConfig stats, Vector2 direction, Vector
      **/
      public List<Entity> createShotGunProjectile (ProjectileConfig stats, Vector2 direction,
                                        Vector2 parentPosition) {
-        Double polarAngle = atan(direction.y / direction.x);
-        float followSpeed = 0.9F;
+
+         direction = direction.nor();
```
Comment (Aryan-S-Gupta) [inline source/core/src/main/com/csse3200/game/entities/factories/ProjectileFactory.java:213]: Please add comments - just for understanding (normalize), Optional

**Review by Aryan-S-Gupta — APPROVED** [review]
Looks Good, suitable rest cases, Really like the implementation of single bullet and multiple having different damage

**Review by manya-k — APPROVED** [review]
Code looks fine. SOMETIMES the melee weapon sound effects are continued even after u stop using.

**Comment by manya-k** [comment]
pls resolve merge conflicts

**Review by vsinh14 — APPROVED** [review]
Looks good

**Comment by manya-k** [comment]
Looks good to be merged .

**Pull request merged**